cpp/cmake_modules/SetupCxxFlags.cmake

```diff
@@ -47,8 +47,8 @@ if(ARROW_CPU_FLAG STREQUAL "x86")
   else()
     set(ARROW_SSE4_2_FLAG "-msse4.2")
     set(ARROW_AVX2_FLAG "-mavx2")
-    # skylake-avx512 consists of AVX512F,AVX512BW,AVX512VL,AVX512CD,AVX512DQ
-    set(ARROW_AVX512_FLAG "-march=skylake-avx512 -mbmi2")
+    # Typical AVX512 subsets consists of AVX512F, AVX512BW, AVX512VL, AVX512CD, AVX512DQ
+    set(ARROW_AVX512_FLAG "-mavx512f -mavx512cd -mavx512vl -mavx512dq -mavx512bw -mbmi2")
```

Comment:
Should we also keep `-march=...` to benefit from modern code selection choices?
(also should we add `-march=something` to other flags above?)

Reply:
According to https://stackoverflow.com/a/55371559/10194, it's recommended to tune for a modern micro-architecture.

Reply:
Yes. Most build use "-march=haswell" for AVX2, "-march=skylake-avx512" for AVX512.

Thus we has to use "-mavx512f -mavx512cd -mavx512vl -mavx512dq -mavx512bw -march=skylake-avx512" for this issue, a little duplicated...